cmd/nri-kubernetes/main.go

```diff
@@ -74,6 +75,17 @@ func main() {
 		}
 	}
 
+	healthServer := health.New(health.WithLogger(logger))
+
+	if c.HealthAddress != "" {
+		go func(address string) {
+			err := healthServer.ListenAndServe(address)
+			if err != nil {
+				logger.Errorf("Error starting health server: %v", err)
+			}
+		}(c.HealthAddress)
+	}
```

Comment:
Nit: maybe the 'health server start goroutine function' could be responsibility of the healthServer itself (just to avoid adding complexity to the main function). Maybe something like this:

```go
if c.HealthAddress != "" {
    healthServer := health.New(health.WithLogger(logger))
    go healtServer.Start(c.HealdAddress)
}
```
The healthServer already has the logger, so error the handling can be done there.

WDYT?

Reply:
Fair point! When I first wrote that the logger was not inside yet, but now it makes total sense.

Reply:
Giving it a second thought... It is not the use case, but if the do it that way, the caller will never have a way to see if the `Server` has errored.
Again we are not checking it here, but from the API design perspective it does not feel 100% right removing that capability.

Reply:
Fair enough! If making the main more complex was a problem we could always use a helper to start the healthServer. 👍